### ISSUE TITLE
Some games are not recognized

### ISSUE THREAD

**Issue by JohnnyDeeee** [adirh3/Sonar.AutoSwitch#2]
When running Call of Duty: Modern warfare 2 the app doesn't recognize it and won't update the Sonar config.
I think this has to do with `Win32WindowEventManager`, i debugged the app and never got in https://github.com/adirh3/Sonar.AutoSwitch/blob/master/Sonar.AutoSwitch/Services/Win32/Win32WindowEventManager.cs#L67

I tried

```cs
_windowEventHook = SetWinEventHook(
				1, // eventMin
				2147483647, // eventMax
				IntPtr.Zero, // hmodWinEventProc
				_lpfnWinEventProc, // lpfnWinEventProc
				0, // idProcess
				0, // idThread
				WINEVENT_OUTOFCONTEXT | WINEVENT_SKIPOWNPROCESS);
```

thinking this would show me all the events, but it never triggered on `cod.exe`.

In my `procexp64` program i also cannot get any information on the process.. Could it be that this is somehow "protected" and need admin privileges to access this process? (running `procexp64` with admin privileges doesn't work, so not sure if this will work in the app..)
![image](https://user-images.githubusercontent.com/9005496/205490790-675721c6-1e81-472f-b7d4-4359ab7163e3.png)

**Comment by adirh3** [comment]
Try running the AutoSwitch as administrator and see if it works

**Comment by thywolf** [comment]
Looks like some games are protected enough to make the Win32WindowEventManager fail when switching to such a game. I'm testing a fallback for such cases - if an exception is raised when switching to such a game, making the original detection impossible, I'm just checking if the process with a given name is running. Not ideal (don't open two problematic games at the same time!), but seems to be working for Valorant :-)

[https://github.com/thywolf/Sonar.AutoSwitch/commit/3eb43303b971fd7be3c13399c53e110e5adea7ff](url)

**Comment by JohnnyDeeee** [comment]
> Not ideal (don't open two problematic games at the same time!)

You could go through the whole list to check if you are running multiple games which both have profiles in the app. When you detect multiple you can show a warning to the user that he must choose which configuration will be the active one (because you can only have 1 active configuration).

**Comment by adirh3** [comment]
@JohnnyDeeee @thywolf added support for matching by title, it should help with these games. 
Check out 28.2.0 release and let me know if it works for you.

**Comment by thywolf** [comment]
Switching now works like a charm for me, @adirh3 ! Thanks a million!

**Comment by FowD8** [comment]
was this figured out? I haven't been able to get it to work with Valorant...

![image](https://user-images.githubusercontent.com/8668035/207113762-2403d784-66c0-4d95-bd48-59888921d769.png)

**Comment by FowD8** [comment]
Just a note to the above, I guess if you're using `Window Title` it uses a wildcard essentially. so if the window is called "VALORANT", you can just put VAL for the Window title, **BUT** make sure you keep Game executable name empty, I think if it has something in there, it's looking for both, and if it's a protected exe, it won't work

**Comment by adirh3** [comment]
> Just a note to the above, I guess if you're using `Window Title` it uses a wildcard essentially. so if the window is called "VALORANT", you can just put VAL for the Window title, **BUT** make sure you keep Game executable name empty, I think if it has something in there, it's looking for both, and if it's a protected exe, it won't work

Keep EXE name empty and title Valorant worked for me.
